base/applications/rapps/appview.cpp

```diff
@@ -2134,30 +2137,71 @@ CApplicationView::RestoreListSelection(const RESTORELISTSELECTION &Restore)
     }
 }
 
-// this function is called when a item of listview get focus.
-// CallbackParam is the param passed to listview when adding the item (the one getting focus now).
+void
+CApplicationView::_UpdateInstallBtn()
+{
+    if (ApplicationViewType == AppViewTypeInstalledApps)
+    {
+        EnableMenuItem(GetMenu(), ID_INSTALL, MF_GRAYED);
+        m_Toolbar->SendMessageW(TB_ENABLEBUTTON, ID_INSTALL, FALSE);
+    }
+    else if (ApplicationViewType == AppViewTypeAvailableApps)
+    {
+        // We shouldn't get there in APPWIZ-mode.
+        ATLASSERT(!m_MainWindow->m_bAppwizMode);
+
+        /* Even if no ListView item is focused at this point, enable
+         * or disable ID_INSTALL if there are selected applications. */
+        BOOL CanInstall = !m_MainWindow->m_Selected.IsEmpty();
+        CanInstall = CanInstall || !!m_ListView->GetSelectedCount();
+        CanInstall = CanInstall || (m_ListView->GetNextItem(-1, LVNI_FOCUSED | LVNI_SELECTED) != -1);
```

Comment:
Why does it care about the listview selection count **and** finding the first selected item? It seems to me only one of these listview cases would be enough. And the item having focus should not matter at all?